imageregistry/v1/types.go

```diff
@@ -195,6 +210,16 @@ type ImageRegistryConfigStorageS3 struct {
 	// Optional, defaults to false.
 	// +optional
 	VirtualHostedStyle bool `json:"virtualHostedStyle"`
+	// trustedCA is a reference to a config map containing a CA bundle. The
+	// image registry and its operator use certificates from this bundle
+	// together with certificates from the system trust bundle to verify S3
+	// server certificates.
+	//
+	// The namespace for the config map referenced by trustedCA is
+	// "openshift-config". The key for the bundle in the config map is
+	// "ca-bundle.crt".
+	// +optional
+	TrustedCA *S3TrustedCASource `json:"trustedCA,omitempty"`
```

Comment:
avoid the pointer for discoverability

Reply:
What kind of discoverability?

Reply:
@deads2k this API doesn't need "discoverability", it should be discoverable only from documentation. On AWS the operator will provide the default configuration that customers shouldn't edit. On other platforms `spec.storage` is empty, so customers won't be able to discover anything from the yaml, they'll need to pick the storage type first.

So if we remove the pointer, it'll hurt semantics without any benefits.